sdl_android/src/main/java/com/smartdevicelink/trace/Mime.java

```diff
@@ -18,48 +22,61 @@ public static String base64Encode(String str) {
 	}
 
 	public static String base64Encode(byte bytesToEncode[]) {
-		return base64Encode(bytesToEncode, 0, bytesToEncode.length);
+		if(bytesToEncode != null){
+			return base64Encode(bytesToEncode, 0, bytesToEncode.length);
+		}
+		return null;
 	}
 
 	public static String base64Encode(byte bytesToEncode[], int offset, int length) {
+		if(bytesToEncode == null){
+			return null;
+		}
+
 		StringBuilder sb = new StringBuilder();
 
 		int		idxin = 0;
 		int		b64idx = 0;
 
-		for (idxin=offset;idxin < offset + length;idxin++) {
-			switch ((idxin-offset) % 3) {
+		try {
+
+			for (idxin = offset; idxin < offset + length; idxin++) {
+				switch ((idxin - offset) % 3) {
+					case 0:
+						b64idx = (bytesToEncode[idxin] >> 2) & 0x3f;
+						break;
+					case 1:
+						b64idx = (bytesToEncode[idxin] >> 4) & 0x0f;
+						b64idx |= ((bytesToEncode[idxin - 1] << 4) & 0x30);
+						break;
+					case 2:
+						b64idx = (bytesToEncode[idxin] >> 6) & 0x03;
+						b64idx |= ((bytesToEncode[idxin - 1] << 2) & 0x3c);
+						sb.append(m_base64Chars.charAt(b64idx));
+						b64idx = bytesToEncode[idxin] & 0x3f;
+						break;
+				}
+				sb.append(m_base64Chars.charAt(b64idx));
+			}
+
+			switch ((idxin - offset) % 3) {
 				case 0:
-					b64idx = (bytesToEncode[idxin] >> 2) & 0x3f;
 					break;
 				case 1:
-					b64idx = (bytesToEncode[idxin] >> 4) & 0x0f;
-					b64idx |= ((bytesToEncode[idxin-1] << 4)& 0x30);
+					b64idx = (bytesToEncode[idxin - 1] << 4) & 0x30;
+					sb.append(m_base64Chars.charAt(b64idx));
+					sb.append("==");
 					break;
 				case 2:
-					b64idx = (bytesToEncode[idxin] >> 6) & 0x03;
-					b64idx |= ((bytesToEncode[idxin-1] << 2)& 0x3c);
+					b64idx = ((bytesToEncode[idxin - 1] << 2) & 0x3c);
 					sb.append(m_base64Chars.charAt(b64idx));
-					b64idx = bytesToEncode[idxin] & 0x3f;
+					sb.append('=');
 					break;
 			}
-			sb.append(m_base64Chars.charAt(b64idx));
-		}
 
-		switch ((idxin-offset) % 3) {
-			case 0:
-				break;
-			case 1:
-				b64idx = (bytesToEncode[idxin-1] << 4) & 0x30;
-				sb.append(m_base64Chars.charAt(b64idx));
-				sb.append("==");
-				break;
-			case 2:
-				b64idx = ((bytesToEncode[idxin-1] << 2)& 0x3c);
-				sb.append(m_base64Chars.charAt(b64idx));
-				sb.append('=');
-				break;
-		}	
-		return sb.toString();
+			return sb.toString();
+		}catch (ArrayIndexOutOfBoundsException e){

```

Comment:
This seems like a pretty easy exception to prevent, we should probably try that rather than just catch it.